sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncClient.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.encryption;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosAsyncClientEncryptionKey;
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.CosmosAsyncDatabase;
+import com.azure.cosmos.implementation.caches.AsyncCache;
+import com.azure.cosmos.models.ClientEncryptionPolicy;
+import com.azure.cosmos.models.CosmosClientEncryptionKeyProperties;
+import com.microsoft.data.encryption.cryptography.EncryptionKeyStoreProvider;
+import reactor.core.publisher.Mono;
+
+/**
+ * CosmosClient with Encryption support.
+ */
+public class CosmosEncryptionAsyncClient {
+    private final CosmosAsyncClient cosmosAsyncClient;
+    private final AsyncCache<String, ClientEncryptionPolicy> clientEncryptionPolicyCacheByContainerId;
+    private final AsyncCache<String, CosmosClientEncryptionKeyProperties> clientEncryptionKeyPropertiesCacheByKeyId;
+    private EncryptionKeyStoreProvider encryptionKeyStoreProvider;
+
+    CosmosEncryptionAsyncClient(CosmosAsyncClient cosmosAsyncClient,
+                                EncryptionKeyStoreProvider encryptionKeyStoreProvider) {
+        if (cosmosAsyncClient == null) {
+            throw new IllegalArgumentException("cosmosClient is null");
+        }
+        if (encryptionKeyStoreProvider == null) {
+            throw new IllegalArgumentException("encryptionKeyStoreProvider is null");
+        }
+        this.cosmosAsyncClient = cosmosAsyncClient;
+        this.encryptionKeyStoreProvider = encryptionKeyStoreProvider;
+        this.clientEncryptionKeyPropertiesCacheByKeyId = new AsyncCache<>();
+        this.clientEncryptionPolicyCacheByContainerId = new AsyncCache<>();
+    }
+
+    public EncryptionKeyStoreProvider getEncryptionKeyStoreProvider() {
+        return encryptionKeyStoreProvider;
+    }
+
+    Mono<ClientEncryptionPolicy> getClientEncryptionPolicyAsync(
+        CosmosAsyncContainer container,
+        boolean shouldForceRefresh) {
```

Comment:
>Refresh) { [](start = 27, length = 10)

Does anyone pass this as true?

Reply:
We are not passing it true as of now. Good to have a capability , and also it will in line with other cache methods .